unlock-app/src/components/interface/keychain/AddToPhoneWallet.tsx

```diff
@@ -1,99 +1,138 @@
 import { Box, Modal, Size } from '@unlock-protocol/ui'
 import QRCode from 'qrcode.react'
-import { createWalletPass, Platform } from '../../../services/ethpass'
-import { useAuth } from '~/contexts/AuthenticationContext'
+import {
+  generateAppleWalletPass,
+  generateGoogleWalletPass,
+} from '../../../services/passService'
 import { ToastHelper } from '~/components/helpers/toast.helper'
+import Image from 'next/image'
 
 interface ApplePassModalProps {
   isOpen: boolean
   setIsOpen: (state: boolean) => void
   applePassUrl?: string
 }
 
-/**
- * Apple needs a modal to show a QR code
- * @param param0
- * @returns
- */
 export const ApplePassModal = ({
   isOpen,
   setIsOpen,
   applePassUrl,
 }: ApplePassModalProps) => {
   return (
     <Modal isOpen={isOpen} setIsOpen={setIsOpen}>
-      {applePassUrl && (
+      {applePassUrl ? (
         <div className="flex flex-col items-center">
           <p>
-            Please scan this device with your phone or{' '}
+            Please scan this QR code with your phone or{' '}
             <a href={applePassUrl} className="underline">
               click here to download it
             </a>
             .
           </p>
           <QRCode value={applePassUrl} size={256} includeMargin />
         </div>
+      ) : (
+        <p>Generating your pass...</p>
       )}
-      {!applePassUrl && <p>Generating your pass...</p>}
     </Modal>
   )
 }
 
 interface AddToWalletProps {
-  children: React.ReactNode
   as: React.ElementType
   network: number
   lockAddress: string
   tokenId: string
-  platform: Platform
   handlePassUrl: (url: string) => void
   disabled?: boolean
   active?: boolean
-  name: string
+  minimised?: boolean
   iconLeft?: JSX.Element
   size?: Size
   variant?: string
   className?: string
 }
 
-export const AddToDeviceWallet = ({
-  children,
+// Add to Apple wallet
+export const AddToAppleWallet = ({
```

Comment:
I am not sure we really need 2 components here. can you expand a bit?

Reply:
no we don't! fixed!